feat: 디자인 리소스 추가 및 담당 구조체 구현

Hansangjin98 opened the pull request:
❗️ 이슈 번호
Closes #6 디자인 리소스 구현

### 📝 작업 유형

- [x] 신규 기능 추가

### 📙 작업 내역

- 이미지 에셋 파일들 추가 
- 폰트 파일들 추가 
- 컬러 에셋 구현 
- DesignKit 모듈에 Font, Color, Image 생성을 담당하는 각각의 구조체 구현
- DesignKit 모듈에 UIViewController의 Gradient 배경색 적용을 위한 Extension 메서드 구현

### 📌 사용 예제

```swift
import DesignKit

let button = UIButton().then {
  $0.setImage(PhoChakImage.createImage(.kakao), for: .normal)
  $0.backgroundColor = PhoChakColor.createColor(.yellow, 400)
  $0.titleLabel?.font = PhoChakFont.createFont(.LargeTitle, 800)
}
```

```swift
import DesignKit

override func viewDidLoad() {
  super.viewDidLoad()

  setGradientBackground()
}
```

<br>
<br>

### 💡 참고사항

- 본 PR에 [PR 템플릿 수정] 작업이 포함되어 있습니다.
- 추후 디자인 변경 사항 발생시 이미지 에셋이 추가되거나 변경될 수 있습니다.

Hansangjin98 committed:
feat: #6 - 디자인 리소스 구현

info.plist에 폰트 등록

Hansangjin98 committed:
feat: #6 - 디자인 리소스 구현

앱 리소스 폴더에 폰트 파일들 추가

Hansangjin98 committed:
feat: #6 - 디자인 리소스 구현

DesignKit 모듈에 Font 구조체 구현

Hansangjin98 committed:
feat: #6 - 디자인 리소스 구현

App Assets에 컬러 에셋들 추가

Hansangjin98 committed:
feat: #6 - 디자인 리소스 구현

DesignKit 모듈에 Color 구조체 구현

Hansangjin98 committed:
feat: #6 - 디자인 리소스 구현

App Assets에 이미지 에셋들 추가

Hansangjin98 committed:
feat: #6 - 디자인 리소스 구현

DesignKit 모듈에 Image 구조체 구현

Hansangjin98 committed:
feat: #6 - 디자인 리소스 구현

열거형 네이밍 Literal로 변경

Hansangjin98 committed:
feat: #6 - 디자인 리소스 구현

DesignKit 모듈에 UIViewController의 Gradient 배경색 적용을 위한 Extension 메서드 구현

Hansangjin98 committed:
docs: PR 템플릿 수정

Hansangjin98 committed:
style: #6 - 디자인 리소스 구현

self 제거, 띄어쓰기 수정

nil-ian commented on Targets/DesignKit/Sources/Font/PhoChakFont.swift:
```diff
@@ -0,0 +1,44 @@
+  public static func createFont(_ size: FontSizeLiteral, _ weight: Int) -> UIFont {
+    var fontName: String {
+      switch weight {
+      case 800: return "Pretendard-ExtraBold"
```
폰트 갯수가 다양하지 않다면 괜찮았을지도 모르겠지만 폰트 종류가 다양하다보니 원하는 폰트를 반환받기 위해 weight값을 다 알고 있어야 하는 구조네ㅠㅠ
font 혹은 weight 타입을 enum으로 정의해놓고 그걸  파라미터로 전달받으면 함수 호출이 더욱 직관적이고 편리할것 같은데 어때??

nil-ian commented on Targets/DesignKit/Sources/Color/PhoChakColor.swift:
```diff
@@ -0,0 +1,29 @@
+  }
+
+  // MARK: Methods
+  public static func createColor(_ color: ColorLiteral, _ weight: Int) -> UIColor {
```
파라미터 인자로 받는 Int 하드코드의 범위가 너무 광범위하다보니 원하는 색상을 뽑아내기 위해서는 매번 피그마에 들어가서 그 색상에 대응되는 weight값을 보고 함수를 호출해야 할 것 같아ㅠㅠ 함수 caller 입장에서 더욱 편하게 사용할 수 있는 방법은 없을까?

nil-ian commented on Targets/DesignKit/Sources/Extensions/UIViewController+.swift:
```diff
@@ -0,0 +1,23 @@
+import UIKit
+
+public extension UIViewController {
+  func setGradientBackground() {
```
- UIView클래스의 layer 프로퍼티에 적용하는거면 UIView의 익스텐션으로 제공하는게 더욱 좋을것 같아~
- 디자인 가이드를 보면 gradient 적용하는 곳이 다양하던데 startColor, endColor를 파라미터로 전달받아서 적용하는 방식으로 하면 범용적으로 사용할 수 있을듯!

nil-ian commented on Targets/DesignKit/Sources/Color/PhoChakColor.swift:
```diff
@@ -0,0 +1,29 @@
+
+import UIKit
+
+public struct PhoChakColor {
```
컬러, 이미지, 폰트 제공이 static을 통해 제공되는데 익스텐션이 아닌 별도 구조체를 정의해서 제공하는 이유가 어떤거야~?
구조체로 정의했지만 구조체 객체를 사용하지는 않는데 `enum`으로 정의해서 제공하는 것과는 어떤 차이가 있을지 한번 고민해보면 좋을듯!

개인적으로는 자체 클래스 내 익스텐션 또는 생성자로 제공 || 익스텐션으로 enum을 정의해서 제공하는게 메소드 체이닝으로 자동완성도 훨씬 간편할것 같아~

nil-ian commented on Targets/DesignKit/Sources/Font/PhoChakFont.swift:
```diff
@@ -0,0 +1,44 @@
+
+import UIKit
+
+public struct PhoChakFont {
```
폰트도 static을 통해 제공되는데 별도 구조체 통해서 제공하는 이유 궁금!

nil-ian commented on Targets/DesignKit/Sources/Image/PhoChakImage.swift:
```diff
@@ -0,0 +1,32 @@
+
+import UIKit
+
+public struct PhoChakImage {
```
이미지도~

nil-ian reviewed (approved):
같이 고민해봅시다~

Hansangjin98 committed:
feat: #6 - UIFont 관련 리뷰 사항 반영

- 구조체 제거 후 Extension 방식으로 변환
- FontWeight 열거형 구현 및 함수 파라미터에 적용

Hansangjin98 committed:
feat: #6 - UIImage 관련 리뷰 사항 반영

- 구조체 제거 후 Extension 방식으로 변환

Hansangjin98 committed:
feat: #6 - UIColor 관련 리뷰 사항 반영

- 구조체 제거 후 Extension 방식으로 변환
- ColorWeight 열거형 구현 및 함수 파라미터에 적용

Hansangjin98 committed:
feat: #6 - UIView 관련 리뷰 사항 반영

- Extension 대상을 UIViewController에서 UIView로 변경
- startColor, endColor 전달받아 View에 Gradient 적용하는 setGradient 메서드 구현
- BaseViewController에 배치할 setGradientBackground 메서드 구현

nil-ian reviewed (approved):
훨씬 사용하기 편리해졌다! 굳굳~
